code/game/objects/items/carp_lasso.dm

```diff
@@ -1,92 +1,133 @@
 /obj/item/mob_lasso
 	name = "space lasso"
-	desc = "Comes standard with every space-cowboy.\nCan be used to tame space carp."
+	desc = "Comes standard with every space-cowboy.\n<span class='notice'>Can be used to tame space carp.</span>"
 	icon = 'icons/obj/carp_lasso.dmi'
 	icon_state = "lasso"
 	///Ref to timer
 	var/timer
 	///Ref to lasso'd carp
 	var/mob/living/simple_animal/mob_target
+	///Range we can lasso things at
+	var/range = 8
 	///Whitelist of allowed animals
-	var/list/whitelist_mobs = list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin)
+	var/list/whitelist_mobs
+	///blacklist of disallowed animals
+	var/list/blacklist_mobs
+
+/obj/item/mob_lasso/Initialize(mapload)
+	. = ..()
+	whitelist_mobs = typecacheof(list(/mob/living/simple_animal/hostile/carp, /mob/living/simple_animal/hostile/carp/megacarp, /mob/living/simple_animal/hostile/carp/lia,\
+	 /mob/living/simple_animal/cow, /mob/living/simple_animal/hostile/retaliate/dolphin), only_root_path = TRUE)
+	blacklist_mobs = list()
```

Comment:
Instead of doing this in initialise, make a new proc that sets the whitelisted and blacklisted procs. Override that proc on child types. This will prevent unnecessary list creation for child types.
Also if you are going to use typecacheof, you need to cache the results. Typecacheof is a very expensive proc and since the checking is being done very infrequently it is probably better to store a list of parents and then use a for loop with istype() checks.

Reply:
Should be done, let me know if I missed anything.